Fix(Redirection on deleting monitor): fixed redirection on deleting monitor

Shubhdeep12 opened the pull request:
Fixes: #59723 

When the filter is applied on crons page (monitor listing page) - then prevUrl gets added to URL. which helps in redirection to the correct URL when the monitor is deleted by user.
<!-- Describe your PR here. -->



<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

davidenwang commented on static/app/views/monitors/components/monitorHeaderActions.tsx:
```diff
@@ -22,11 +23,19 @@ type Props = {
+    browserHistory.push(
+      normalizeUrl(
+        `/organizations/${orgId}/crons/${
+          router.location.query.prevUrl ? `?project=${router.location.query.prevUrl}` : ''
```
This method works, but adds unnecessary query parameters to the URL

It also only functions when coming from the monitor overview page, if you were to add a monitor and then delete it immediately you would be redirected back to a page without a project selection

What I think we could do here is leverage the `selection` from `usePageFilters` which holds global state around which project(s) were selected and feed those into the query params. Let me know if that makes sense and if it works out, I haven't tried it.

Shubhdeep12 replied:
Yes @davidenwang correct. Using prevUrl is not needed we need selected filters - projects and environments which will be present in usePageFilters.

And, yes i only covered just one flow but the creation, edition redirection should also be handled and also the breadcrumbs.

I am pushing a commit in sometime for this.

Thanks

davidenwang reviewed (commented):
Thanks for this contribution, left a comment

Shubhdeep12 commented:
> Thanks for this contribution, left a comment

Hi @davidenwang updated the changes. Please check. 

- used selection from usePageFilters to get the applied filters.
- Fixed redirection in Breadcrumbs
- Fixed redirection when detail page opened from create/edit flow.

davidenwang reviewed (approved):
apologies on the late review, but this is looking good. Thanks for working on this!
